spark/src/main/scala/org/apache/spark/sql/delta/util/PartitionUtils.scala

```diff
@@ -117,7 +117,8 @@ object PartitionSpec {
 
 private[delta] object PartitionUtils {
 
-  val timestampPartitionPattern = "yyyy-MM-dd HH:mm:ss[.S]"
+  lazy val timestampPartitionPattern = "yyyy-MM-dd HH:mm:ss[.S]"
+  lazy val utcFormatter = TimestampFormatter("yyyy-MM-dd HH:mm:ss.SSSSSSz", ZoneId.of("UTC"))
```

Comment:
Normalized partition values will be in microseconds, similar to what Kernel is doing https://github.com/delta-io/delta/blob/573a57f62918d0cb8937ca8c9b4047f8d696cc9c/kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/PartitionUtils.java#L45